Assets/_BForBoss/_Core/Scripts/Managers/GunRangeWorldManager.cs

```diff
@@ -20,7 +20,13 @@ public class GunRangeWorldManager : BaseWorldManager
         protected override void Start()
         {
             base.Start();
-            _weaponsManager.Initialize(new CharacterMovementWrapper(_player));
+            _weaponsManager.Initialize(
+                () => _player.CharacterVelocity,
+                () => _player.CharacterMaxSpeed,
+                () => _player.IsWallRunning,
+                () => _player.IsGrounded,
+                () => _player.IsSliding,
+                () => _player.IsDashing);
```

Comment:
Basically, instead of a wrapper in a middle of _Core, I'm passing in what WeaponsManager needs.

Originally these values are only needed by weaponsManager, but now it's also needed by Character. So instead of making a copy, i'm moving it back to Character, and just passing in what the other's need. 

This is Dependency inversion.